### PR TITLE
zellij: 0.5.1 -> 0.6.0

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Motivation for this change
Upgrade to [0.6.0](https://github.com/zellij-org/zellij/releases/tag/v0.6.0).
Add myself as co-maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/zellij/default.nix [hunk 1]
```diff
@@ -22,13 +22,17 @@ rustPlatform.buildRustPackage rec {
   '';
 
   postInstall = ''
-    installShellCompletion assets/completions/zellij.{bash,fish} --zsh assets/completions/_zellij
+    mkdir -p assets/completions
+    for shell in fish bash zsh; do
+      $out/bin/zellij generate-completion $shell > assets/completions/zellij.$shell
+    done
+    installShellCompletion assets/completions/zellij.{bash,fish} --zsh assets/completions/zellij.zsh
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/zellij/default.nix:29]:
```suggestion
    installShellCompletion --cmd \
      --bash <($out/bin/zellij generate-completion bash) \
      --fish <($out/bin/zellij generate-completion fish) \
      --zsh assets/completions/zellij.zsh
```

File: pkgs/tools/misc/zellij/default.nix [hunk 2]
```diff
@@ -22,13 +22,17 @@ rustPlatform.buildRustPackage rec {
...
   '';
 
   meta = with lib; {
     description = "A terminal workspace with batteries included";
     homepage = "https://zellij.dev/";
     license = with licenses; [ mit ];
-    maintainers = with maintainers; [ therealansh ];
+    maintainers = with maintainers; [ therealansh _0x4A6F ];
```
Comment (happysalada) [inline pkgs/tools/misc/zellij/default.nix:36]: This is a tiny thing, but usually, updating a package and adding a maintainer is done in a separate commit.

File: pkgs/tools/misc/zellij/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/tools/misc/zellij/default.nix:29]: I think this one needs a little tweak, since the assets/completions/zellij.zsh needs to be created. But the idea of trying to do it in a single command is good I think.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 121201` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zellij</li>
  </ul>
</details>
